src/avram/database.cr

```diff
@@ -154,16 +155,24 @@ abstract class Avram::Database
     Avram::Connection.new(url, database_class: self.class)
   end
 
-  private def db : DB::Database
+  def db : DB::Database
     @@db ||= @@lock.synchronize do
       # check @@db again because a previous request could have set it after
       # the first time it was checked
       @@db || connection.open
     end
   end
 
+  def current_connection : DB::Connection
+    connections[current_object_id] ||= db.checkout
+  end
+
   private def current_transaction : DB::Transaction?
-    transactions[Fiber.current.object_id]?
+    current_connection.stack.last?
+  end
+
+  def current_object_id : FiberId
+    @@lock_id || Fiber.current.object_id
```

Comment:
I'm not sure this was required for this specific work so far. It will be more useful in the future when we're testing an app that will be running on a different fiber. We can control which connections are used by doing this